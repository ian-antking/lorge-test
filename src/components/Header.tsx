import React from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'

const Logo = styled(Link)`
  font-size: 2em;
  color: ${(props) => props.theme.background};
  font-weight: bold;
  text-decoration: none;
`

const HeaderContainer = styled.header`
  background-color: ${(props) => props.theme.text};
  padding: 10px;
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  align-items: center;
`

const Header: React.FC = (): React.ReactElement => (
  <HeaderContainer>
    <Logo to="/">lÃ¶rge</Logo>
  </HeaderContainer>
)

export default Header
