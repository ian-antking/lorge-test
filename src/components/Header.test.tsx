import { render, fireEvent } from '@testing-library/react'
import { createMemoryHistory } from 'history'
import {
  MemoryRouter,
  unstable_HistoryRouter as HistoryRouter
} from 'react-router-dom'
import Header from './Header'

import '@testing-library/jest-dom'

describe('Header', () => {
  it('renders the website name', () => {
    const { getByText } = render(<Header />, { wrapper: MemoryRouter })

    expect(getByText('lörge')).toBeInTheDocument()
  })

  it('updates history to home when clicked', () => {
    const history = createMemoryHistory({
      initialEntries: ['/not-home']
    })

    const { getByText } = render(
      <HistoryRouter history={history}>
        <Header />
      </HistoryRouter>
    )

    fireEvent.click(getByText('lörge'))

    expect(history.location.pathname).toBe('/')
  })

  it('renders consistently', () => {
    const { asFragment } = render(<Header />, { wrapper: MemoryRouter })

    expect(asFragment()).toMatchSnapshot()
  })
})
