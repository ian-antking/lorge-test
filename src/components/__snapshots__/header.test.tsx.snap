// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders consistently 1`] = `
<DocumentFragment>
  <header
    class="sc-gsnTZi kWdnON"
  >
    <a
      class="sc-bczRLJ gXuhJw"
      href="/"
    >
      l√∂rge
    </a>
  </header>
</DocumentFragment>
`;
